pkg/controller/networkpolicy/controller_querier_replier.go

```diff
@@ -20,119 +20,122 @@ package networkpolicy
+	_, err := eq.networkPolicyController.podInformer.Lister().Pods(namespace).Get(podName)
 	if err != nil {
 		return &EndpointQueryResponse{
 			Endpoints: nil,
-			Error:     err,
-		}
+		}, err
 	}
-	// grab list of all policies from internalNetworkPolicyStore
-	internalPolicies := eq.internalNetworkPolicyStore.List()
 	// create network policies categories
-	applied, _, _ := make([]antreatypes.NetworkPolicy, 0), make([]antreatypes.NetworkPolicy, 0),
-		make([]antreatypes.NetworkPolicy, 0)
-	// filter all policies into appropriate groups
-	for _, policy := range internalPolicies {
-		antreaPolicy := policy.(*antreatypes.NetworkPolicy)
-		println(antreaPolicy.Name)
-		for _, key := range antreaPolicy.AppliedToGroups {
-			// Check if policy is applied to endpoint
-			//TODO: what is this boolean. what is this error?
-			appliedToGroupInterface, _, _ := eq.appliedToGroupStore.Get(key)
-			appliedToGroup := appliedToGroupInterface.(*antreatypes.AppliedToGroup)
-			// if appliedToGroup selects pod in namespace, append policy to applied category
-			for _, podSet := range appliedToGroup.PodsByNode {
-				for _, member := range podSet {
-					trialPodName, trialNamespace := member.Pod.Name, member.Pod.Namespace
-					println(trialPodName)
-					if podName == trialPodName && namespace == trialNamespace {
-						applied = append(applied, *policy.(*antreatypes.NetworkPolicy))
+	applied, ingress, egress := make([]*antreatypes.NetworkPolicy, 0), make([]*antreatypes.NetworkPolicy, 0),
+		make([]*antreatypes.NetworkPolicy, 0)
+	// get all appliedToGroups using pod index, then get applied policies using appliedToGroup
+	appliedToGroups, err := eq.networkPolicyController.appliedToGroupStore.GetByIndex(store.PodIndex, podName + "/" + namespace)
+	if err != nil {
+		return &EndpointQueryResponse{
+			Endpoints: nil,
+		}, err
+	}
+	for _, appliedToGroup := range appliedToGroups {
+		policies, err := eq.networkPolicyController.internalNetworkPolicyStore.GetByIndex(store.AppliedToGroupIndex,
+			string(appliedToGroup.(*antreatypes.AppliedToGroup).UID))
+		if err != nil {
+			return &EndpointQueryResponse{
+				Endpoints: nil,
+			}, err
+		}
+		for _, policy := range policies {
+			applied = append(applied, policy.(*antreatypes.NetworkPolicy))
+		}
+	}
+	// get all addressGroups using pod index, then get ingress and egress policies using addressGroup
+	addressGroups, err := eq.networkPolicyController.addressGroupStore.GetByIndex(store.PodIndex, podName + "/" + namespace)
+	if err != nil {
+		return &EndpointQueryResponse{
+			Endpoints: nil,
+		}, err
+	}
+	for _, addressGroup := range addressGroups {
+		policies, err := eq.networkPolicyController.internalNetworkPolicyStore.GetByIndex(store.AddressGroupIndex,
+			string(addressGroup.(*antreatypes.AddressGroup).UID))
+		if err != nil {
+			return &EndpointQueryResponse{
+				Endpoints: nil,
+			}, err
+		}
+		for _, policy := range policies {
+			for _, rule := range policy.(*antreatypes.NetworkPolicy).Rules {
```

Comment:
The index is useful here, so don't ignore it with `_`. Hopefully it is guaranteed to match the order in the original K8s NetworkPolicy spec.

Reply:
How should I utilize the index?